x-pack/plugins/enterprise_search/server/utils/create_ml_inference_pipeline.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+import { formatMlPipelineBody } from './create_pipeline_definitions';
+
+export interface CreatedPipeline {
+  created: string;
+}
+
+// TODO: document
+export const createMlInferencePipeline = async (
+  pipelineName: string,
+  modelId: string,
+  sourceField: string,
+  destinationField: string,
+  esClient: ElasticsearchClient
+): Promise<CreatedPipeline> => {
+  const inferencePipelineGeneratedName = `ml-inference-${pipelineName}`;
+
+  // Check for existing pipeline
+  let exists = false;
+  try {
+    await esClient.ingest.getPipeline({ id: inferencePipelineGeneratedName });
+    exists = true;
+  } catch (err) {
+    // NOP
+  }
```

Comment:
Not sure if this is the cleanest way to handle the existing pipeline scenario. `getPipeline()` throws a 404 if the pipeline doesn't exist.

Reply:
That's a shame that there isn't an easy way to list the pipelines filtered by name... :/ You could just [get all the pipelines](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-pipeline-api.html) and filter yourself, but that seems very wasteful - so, the way you have it here is probably the best for now...

I would suggest however checking the response - if it's a 404 or a 200 / OK, cool, we know what to do. If it's another error code, I'd bubble that up.

Reply:
BTW - I found [this code in Kibana that handles checking for an existing pipeline name](https://github.com/elastic/kibana/blob/0073a064b7831f28febaf02759ab122e52bf1534/x-pack/plugins/ingest_pipelines/server/routes/api/create.ts)... we should probably follow the same pattern that exists here...

Reply:
One more comment here as well - instead of setting the `exists = true;` and checking it / returning later - best to just do the return right there inside the `try` (on line 30) - always best to return early if possible...

Reply:
> BTW - I found [this code in Kibana that handles checking for an existing pipeline name](https://github.com/elastic/kibana/blob/0073a064b7831f28febaf02759ab122e52bf1534/x-pack/plugins/ingest_pipelines/server/routes/api/create.ts)

Great find, thanks! That said I'm not a fan of mixing concerns (API and business logic in this case). Would it be OK if I do it like this in my code:
- `create_ml_inference_pipeline.ts`: check for pipeline, if exists: return some object that indicates this (e.g. `{ "exists": true, "id": ...}`).
- `indices.ts` based on the return payload from `create_ml_inference_pipeline.ts`, sends the corresponding response.

Reply:
Makes sense... I'm all for more modularity where it makes sense...